### ISSUE TITLE
Overwolf? [closed]

### ISSUE THREAD

**Issue by cherob** [TheHolyWaffle/TeamSpeak-3-Java-API#198]
How i can track if Client use Overwolf.

**Comment by rogermb** [comment]
No 100% certain, but there unfortunately doesn't seem to be a way to check that by using the server query interface.

It might be a better idea to ask that question on the TeamSpeak forums, though, as we only provide a (thin) wrapper around their server query interface.

**Comment by Kakifrucht** [comment]
I reviewed the contents of the Client map, which you can get via ``Client.getMap()``, and it contains the key ``client_badges``. If overwolf is used, it will contain ``overwolf=1`` as value. Should we add a ClientProperty for easier lookup?

**Comment by rogermb** [comment]
> I reviewed the contents of the Client map, which you can get via Client.getMap(), and it contains the key client_badges. If overwolf is used, it will contain overwolf=1 as value.

That's exciting, I didn't know about that property yet! 😄 
We don't even have a getter for it on the `Client` / `ClientInfo` object, either.

> Should we add a ClientProperty for easier lookup?

Yeah, the path forward will be:

1. Add a ClientProperty for `client_badges`
2. Figure out when exactly this property is included (`clientinfo`? `clientlist` without any special flags? `clientlist` with flags?)
3. Figure out what the exact format of the value of `client_badges` is - most likely we need to parse it as just a `Map<String, String>`.
4. Finally, add a getter to `Client` / `ClientInfo` that does that parsing

So much stuff to do, so little time...

**Comment by cherob** [comment]
I got it:
`	
public static void updateOverwolf(int did, int id){
		String overwolf = api.getClientInfo(id).get("client_badges");
		overwolf = overwolf.replace("overwolf=" , "");
		overwolf = overwolf.replace("Overwolf=" , "");
		int ow = Integer.parseInt(overwolf);
		System.out.println(ow);
		if(ow == 1 || overwolf.contains("1")){
                       //OVERWOLF
		}else{
                       //NO OVERWOLF
		}
	}
`

**Comment by Kakifrucht** [comment]
Keep in mind that badges can contain more than just the overwolf badge.

**Comment by cherob** [comment]
I change it to:

``` Java	
public static void updateOverwolf(int did, int id){
		String overwolf = api.getClientInfo(id).get("client_badges");
		int ow = Integer.parseInt(overwolf);
		System.out.println(ow);
		if(overwolf.contains("overwolf=1")){
                       //OVERWOLF
		}else{
                       //NO OVERWOLF
		}
	}
```

**Comment by Kakifrucht** [comment]
I don't believe the changes were implemented yet?

**Comment by rogermb** [comment]
They have been implemented in 4a02ee487e75e0dcade2de169da9e4e4fd59ea94 over on 1.1-dev, but you're right, this commit has not been shipped in a release yet, which is when we usually close these kinds of issues.

**Comment by Kakifrucht** [comment]
Whoops must have skipped it when checking the commit history.
